### PR TITLE
Parallelization and recount2 analysis

### PR DESCRIPTION
**Pull request by ajlee21**
This PR has 2 main changes despite the large number of new files.  These new files are largely duplicates of previous scripts but tailored for the recount2 dataset as opposed to the previous Pseudomonas dataset.

Files to review for the recount2 analysis are:
1.  scripts/analysis_2/5_create_figs_manuscript.ipynb which shows the results of the sample-level simulation using a subset of the recount2 dataset
2.  scripts/analysis_3/5_create_figs_manuscript.ipynb which shows the results of the experiment-level simulation using a subset of the recount2 dataset
3.  scripts/analysis_2/0_download_data.R script used to download the recount2 dataset and get a random subset of it to use

All the scripts are now parallelized in order to run the simulation using multiple random seeds in order to generate a confidence interval.  
An example of the one-off simulations can be found in arachive/scripts/analysis_0_single_run/  and an example of the the updated parallized format can be found in scripts/analysis_0/2_sim_experiment_uncorrected.ipynb

The next PR will be to clean up this repository to make it publication ready.

### CODE REVIEW

File: scripts/analysis_2/0_download_data.R [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+# Sample: Physical biospecimen on which sequencing was performed, biological source material (i.e. HeLa cell line).  
+#         A project contains many samples
+# Experiment: Unique sequencing library for a specific sample.  A sample can have multiple experiments, most have 1 experiment.
+# Run: Sequencing run.  An experiment can contain many runs (i.e. technical replicates)
+# In this case, we want to group runs into projects (for "experiment-level" simulation)
+project_ids <- unique(metadata$project)
+
+# Entire recount2 is 8TB
+# We will only select the top 10 studies instead
+selected_project_ids <- sample(project_ids, 50)
```
Comment (danich1) [inline scripts/analysis_2/0_download_data.R]: > We will only select the top 10 studies instead

^Your comment says select top 10 but code reads sample fifty project ids. Am I reading this correctly?

**Comment by danich1** [comment]
Also design comment:

```python
panel_C = ggplot(all_corrected_data_df[all_corrected_data_df['Comparison'] != '1'],
                 aes(x='PC1', y='PC2')) \
    + geom_point(aes(color='No. of experiments'), 
                 alpha=0.1) \
    + facet_wrap('~Comparison') \
    + labs(x = "PC 1",
           y = "PC 2", 
           title = "PCA of experiment 1 vs multiple experiments") \
    + theme_bw() \
    + theme(
        legend_title_align = "center",
        plot_background=element_rect(fill='white'),
        legend_key=element_rect(fill='white', colour='white'), 
        plot_title=element_text(weight='bold')
    )\
    + guides(colour=guide_legend(override_aes={'alpha': 1})) \
    + scale_color_manual(['#bdbdbd', '#1976d2']) \
    + geom_point(data=all_corrected_data_df[all_corrected_data_df['Comparison'] == '1'],
                 alpha=0.1, 
                 color='#bdbdbd')
```

You can surround multiple lines with parenthesis instead of using backslash.  Example below:
```python
panel_C = (
ggplot(all_corrected_data_df[all_corrected_data_df['Comparison'] != '1'],
                 aes(x='PC1', y='PC2')) 
+ geom_point(aes(color='No. of experiments'), alpha=0.1) 
+ facet_wrap('~Comparison') 
+ labs(x = "PC 1",
           y = "PC 2", 
           title = "PCA of experiment 1 vs multiple experiments") 
    + theme_bw() 
    + theme(
        legend_title_align = "center",
        plot_background=element_rect(fill='white'),
        legend_key=element_rect(fill='white', colour='white'), 
        plot_title=element_text(weight='bold')
    )
    + guides(colour=guide_legend(override_aes={'alpha': 1})) 
    + scale_color_manual(['#bdbdbd', '#1976d2']) 
    + geom_point(data=all_corrected_data_df[all_corrected_data_df['Comparison'] == '1'],
                 alpha=0.1, 
                 color='#bdbdbd')
)
```

**Comment by ajlee21** [comment]
```In scripts/analysis_3/5_create_figs_manuscript.ipynb why is the score for the batch corrected dataset at 50 partitions so low (plot with dark blue points)? Is it that the batch correction is removing signal rather than noise?

^ same questions for scripts/analysis_2/5_create_figs_manuscript.ipynb```

You're exactly correct.  The batch correction is removing the biological signal that we are interested in, instead of the noise.  That is why if you look at the PCA plots, the dark blue points overlap a very small region of the grey points (original gene expression data)

```
Just noticed that the uncorrelated score for uncorrected svcca in analysis_2 is lower than in analysis_3. Any insight on this? I'd expect it to be part of the data manipulation issue, but curious to know why.```

This is a very good observation!  For both the Pseudomonas and recount2 datasets, there is less of a drop in similarity when simulating at the experiment level compared to the sample level.  This is something that we are still thinking about hypotheses for, so I don't have an answer yet.

Thank you for the tip about the formatting, I will add this when i go through and clean up the scripts in the next PR

File: scripts/analysis_2/0_download_data.R (hunk 1 above)
Comment (ajlee21) [reply scripts/analysis_2/0_download_data.R]: This is a typo, I started with just 10 when i was first writing these scripts to test.  I will change this

**Pull request merged**